File: HelpSoftware/Help/Help.Web/Web/src/app/EF/UserLikedArticel.ts
import { User, Article } from "./EF";

export class UserLikedArticel {
  ID_UserLikedArticel: number;
  LikedUser: User;
  LikedArticel: Article;
  LikedTime: Date;

  deserialize(input) {
    this.ID_UserLikedArticel = input.ID_UserLikedArticel;
    this.LikedUser = new User().deserialize(input.LikedUser);
    this.LikedArticel = new Article().deserialize(input.LikedArticel);
    this.LikedTime = input.Date;
    return this;
  }
}

File: HelpSoftware/Help/Help.EF/Entitys/UserLikedArticel.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Help.EF
{
  public class UserLikedArticel : IHelpEntity
  {
    [Key]
    public virtual int ID_UserLikedArticel { get; set; }

    public virtual User LikedUser { get; set; }

    public virtual Article LikedArticel { get; set; }

    public virtual DateTime LikedTime { get; set; }

    [Timestamp]
    public virtual byte[ ] TimeStamp { get; set; }

    public string GetEntityName()
    {
      return "UserLikedArticel";
    }

    public int GetID()
    {
      return ID_UserLikedArticel;
    }

    public bool GetWriteHistoryEntry()
    {
      return true;
    }
  }
}